addon/contentScript.js

```diff
@@ -1,26 +1,23 @@
 /** React to commands sent from the background script. */
+
+function labelledSuccessFunction(found, traineeId) {
+    // Assume the out() key and the data-fathom attr are both identical
+    // to the key of the trainee in the map.
+    // Using the first found node is arbitrary.
+    return found[0].element.dataset.fathom === traineeId;
+}
+
 async function dispatch(request) {
-    switch (request.type) {
-        case 'rulesetSucceeded':
-            // Run the trainee ruleset of the given ID with the given coeffs
-            // over the document, and report whether it found the right
-            // element.
-            const rules = trainees.get(request.traineeId).rulesetMaker(request.coeffs);
-            const facts = rules.against(window.document);
-            // Assume the out() key and the data-fathom attr are both identical
-            // to the key of the trainee in the map.
-            const found = facts.get(request.traineeId);
-            if (found.length >= 1) {
-                const fnode = found[0];  // arbitrary pick
-                if (fnode.element.getAttribute('data-fathom') === request.traineeId) {
-                    return true;
-                }
-                //console.log(urlFilename(window.location.href), ": found wrong answer class=", fnode.element.getAttribute('class'), 'id=', fnode.element.getAttribute('id'));
-            } else {
-                //console.log(urlFilename(window.location.href), ": found nothing.");
-            }
-            return false;
-            break;  // belt, suspenders
+    if (request.type === 'rulesetSucceeded') {
+        // Run the trainee ruleset of the given ID with the given coeffs
+        // over the document, and report whether it found the right
+        // element.
+        const trainee = trainees.get(request.traineeId);
+        const rules = trainee.rulesetMaker(request.coeffs);
+        const successFunc = trainee.successFunction || labelledSuccessFunction;
+        const facts = rules.against(window.document);
+        const found = facts.get(request.traineeId);
```

Comment:
If we move this and the next line into labeledSuccessFunction(), we accomplish https://github.com/mozilla/fathom-fox/issues/7 too! The way I envision it, a set of trainees that address all the out()s of a multi-out ruleset could like like...

```js
trainees.set(
    'overlay',
    {
        coeffs: [2, 1, 3, 1, 1],
        rulesetMaker: commonRuleset
    }
);

trainees.set(
    'popup',
    {
        coeffs: [1, 9, 4],
        rulesetMaker: commonRuleset
    }
);

trainees.set(
    'closeButton',
    {
        coeffs: [1, 1, 1],
        rulesetMaker: commonRuleset
    }
);
```

Since you're already passing the trainee ID into the success function, those examples can just use the default one.

Note that I haven't actually tried this, so there may be a fatal flaw here. :-)